blockchain/api.go

```diff
@@ -696,22 +696,27 @@ func (api *BasicEventsAPI) GetEvents(ctx context.Context, fromBlockInitial *big.
 					api.logger.Info("no logs, skipping")
 					continue
 				}
-				fromBlock = big.NewInt(int64(logs[numLogs-1].BlockNumber))
-
-				var eventTS uint64
-				for _, log := range logs {
-					logBlockNumber := big.NewInt(int64(log.BlockNumber))
-					if lastBlockNumber.Cmp(logBlockNumber) != 0 {
```

Comment:
Looks like the problem was in "==" comparision instead of ">=". I don't get the idea of adding some strange double check.

Reply:
there was no check in the first place, events were emitted from logs unconditionally